DOCKER_TLS_VERIFY usage differs from cli

with the usual TLS settings docker ps works fine:
```
DOCKER_HOST=tcp://192.168.59.103:2376 \
DOCKER_CERT_PATH=~/.boot2docker/certs/boot2docker-vm \
DOCKER_TLS_VERIFY=1 \
  docker ps
```

but using the same ENV vars with the **NewEnvClient()** throws `x509: certificate signed by unknown authority`

```
DOCKER_HOST=tcp://192.168.59.103:2376 \
DOCKER_CERT_PATH=~/.boot2docker/certs/boot2docker-vm \
DOCKER_TLS_VERIFY=1 \
  go run main.go

panic: An error occurred trying to connect: Get https://192.168.59.103:2376/containers/json?all=1&limit=0: x509: certificate signed by unknown authority
```

Im using OSX with boot2docker 1.9.1

## workaround

Unsetting  `DOCKER_TLS_VERIFY` makes the sample go program fine, but than `docker ps` fails with:
```
Get http://192.168.59.103:2376/v1.21/containers/json: malformed HTTP response "\x15\x03\x01\x00\x02\x02".
* Are you trying to connect to a TLS-enabled daemon without TLS?
```

## sample code

Contents of `main.go`:
```
package main

import (
        "fmt"
        "github.com/docker/engine-api/client"
        "github.com/docker/engine-api/types"
)

func main() {
        cli, err := client.NewEnvClient()
        if err != nil {
                panic(err)
        }

        options := types.ContainerListOptions{All: true}
        containers, err := cli.ContainerList(options)
        if err != nil {
                panic(err)
        }

        for _, c := range containers {
                fmt.Println(c.ID)
        }
}
```

--- a/client/client.go
+++ b/client/client.go
@@ -10,6 +10,8 @@ import (
 	"path/filepath"
 	"strings"
 	"time"
+
+	"github.com/docker/go-connections/tlsconfig"
 )
 
 // Client is the API client that performs all operations
@@ -41,15 +43,17 @@ type Client struct {
 func NewEnvClient() (*Client, error) {
 	var transport *http.Transport
 	if dockerCertPath := os.Getenv("DOCKER_CERT_PATH"); dockerCertPath != "" {
-		tlsc := &tls.Config{}
-
-		cert, err := tls.LoadX509KeyPair(filepath.Join(dockerCertPath, "cert.pem"), filepath.Join(dockerCertPath, "key.pem"))
+		options := tlsconfig.Options{
+			CAFile:             filepath.Join(dockerCertPath, "ca.pem"),
+			CertFile:           filepath.Join(dockerCertPath, "cert.pem"),
+			KeyFile:            filepath.Join(dockerCertPath, "key.pem"),
+			InsecureSkipVerify: os.Getenv("DOCKER_TLS_VERIFY") == "",
+		}
+		tlsc, err := tlsconfig.Client(options)
 		if err != nil {
-			return nil, fmt.Errorf("Error loading x509 key pair: %s", err)
+			return nil, err
 		}
 
-		tlsc.Certificates = append(tlsc.Certificates, cert)
-		tlsc.InsecureSkipVerify = os.Getenv("DOCKER_TLS_VERIFY") == ""
 		transport = &http.Transport{
 			TLSClientConfig: tlsc,
 		}

